### PR TITLE
Implement JSONv2 and XMLv2 output for TimeSeries data.

### PR DESCRIPTION
**Pull request by DanielTOsborne**
This is an implementation of paginated timeseries data, using a new JSONv2 format.

Here's an example of the format output:
```json
{
  "begin" : "2021-05-24T10:00:00-0700[PST8PDT]",
  "end" : "2021-06-08T04:00:00-0700[PST8PDT]",
  "interval" : "PT1H",
  "interval-minutes" : 60,
  "name" : "Black Butte.Flow-Reservoir Inflow.Inst.1Hour.0.NWS-forecast-rev",
  "next-page" : null,
  "office-id" : "SPK",
  "page" : "MTYyMTg3NTYwMDAwMHx8LTE=",
  "page-size" : -1,
  "units" : "cfs",
  "value-columns" : [ {
    "name" : "date-time",
    "ordinal" : 1,
    "datatype" : "java.sql.Timestamp"
  }, {
    "name" : "value",
    "ordinal" : 2,
    "datatype" : "double"
  }, {
    "name" : "quality-code",
    "ordinal" : 3,
    "datatype" : "int"
  } ],
  "values" : [
    [ 1621875600000, 30.0, 0 ], 
    [ 1621879200000, 30.0, 0 ], 
    [ 1621882800000, 30.0, 0 ]
  ]
}
```

XML is also included:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<timeseries>
    <page>MTYyMTg3NTYwMDAwMHx8LTE=</page>
    <pageSize>-1</pageSize>
    <begin>2021-05-24T10:00:00-07:00[PST8PDT]</begin>
    <end>2021-06-08T04:00:00-07:00[PST8PDT]</end>
    <interval>PT1H</interval>
    <name>Black Butte.Flow-Reservoir Inflow.Inst.1Hour.0.NWS-forecast-rev</name>
    <officeId>SPK</officeId>
    <units>cfs</units>
    <valueColumns>
        <column>
            <name>dateTime</name>
            <ordinal>1</ordinal>
            <datatype>java.sql.Timestamp</datatype>
        </column>
        <column>
            <name>value</name>
            <ordinal>2</ordinal>
            <datatype>double</datatype>
        </column>
        <column>
            <name>qualityCode</name>
            <ordinal>3</ordinal>
            <datatype>int</datatype>
        </column>
    </valueColumns>
    <values>
        <record>
            <dateTime>1621875600000</dateTime>
            <value>30.0</value>
            <qualityCode>0</qualityCode>
        </record>
        <record>
            <dateTime>1621879200000</dateTime>
            <value>30.0</value>
            <qualityCode>0</qualityCode>
        </record>
        <record>
            <dateTime>1621882800000</dateTime>
            <value>30.0</value>
            <qualityCode>0</qualityCode>
        </record>
    </values>
</timeseries>
```

### CODE REVIEW

File: cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java [hunk 1]
```diff
@@ -88,10 +107,16 @@ public void getAll(Context ctx) {
...
+            // The following parameters are only used for jsonv2
+            String cursor = ctx.queryParam("cursor",String.class,ctx.queryParam("page",String.class,"").getValue()).getValue();
+            int pageSize = ctx.queryParam("pageSize",Integer.class,ctx.queryParam("pagesize",String.class,Integer.toString(defaultPageSize)).getValue()).getValue();
 
+            String acceptHeader = ctx.header("Accept");
+            ContentType contentType = Formats.parseHeaderAndQueryParm(acceptHeader, null);
 
             switch(format){
                 case "json": {ctx.contentType(Formats.JSON); break;}
+                case "jsonv2": {ctx.contentType(Formats.JSONV2); break;}
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java]: This isn't necessary here, and new formats shouldn't be handled in the query param. See the OfficeController for what we're moving to.

File: cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java [hunk 2]
```diff
@@ -93,8 +98,8 @@ public static void main(String[] args){
         })
         .exception(Exception.class, (e,ctx) -> {
             ctx.status(500);
-            ctx.json("There was an error processing your request");
-            logger.log(Level.WARNING,"error on request: " + ctx.req.getRequestURI(), e);
+            ctx.json(new String[]{"There was an error processing your request", sanitizer.sanitize(e.getMessage())});
```
Comment (MikeNeilson) [inline cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java]: Don't return full exceptions back to the user. I'm thinking we should probably return an random key we also put in the log so it's easier to search.

Comment (DanielTOsborne) [reply cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java]: This is the the standalone instance. Is that not just for debugging/testing?

Comment (MikeNeilson) [reply cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java]: nope, that's in play in other system and depending on future configurations could be production somewhere.

File: cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java (hunk 1 above)
Comment (DanielTOsborne) [reply cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java]: Fair enough. I did add the accept-header method. I added the above so it's easy to see in a browser, but I can remove it.

File: cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java (hunk 2 above)
Comment (DanielTOsborne) [reply cwms_radar_standalone/src/main/java/mil/army/usace/hec/RadarAPI.java]: Ah, I also added pretty JSON output in the standalone instance. Would you suggest I remove that too?

File: cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java [hunk 3]
```diff
@@ -96,7 +96,7 @@ public void getAll(Context ctx) {
                 case "wml2": {ctx.contentType(Formats.WML2); break;}
                 case "png":
                 case "jpg":{
-                    ctx.contentType("text/plain");
+                    ctx.contentType(Formats.PLAIN);
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/api/LevelsController.java:99]: Thanks for cleaning these up.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/CwmsDTOPaginated.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import io.swagger.v3.oas.annotations.media.Schema;
+
+/***
+ * Provides a framework for a paginated result set.
+ * Implementation details are subject to inheritors.
+ * @author Daniel Osborne
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlAccessorOrder(XmlAccessOrder.ALPHABETICAL)
+public abstract class CwmsDTOPaginated implements CwmsDTO {
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/CwmsDTOPaginated.java:25]: Making this one class all paginated things derived from was a great idea. Thanks.

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**